Programs_Training/DtPC_visualizer.py

```diff
@@ -41,59 +41,78 @@ def matrix_extrinsic(vector_array, rot_matrix):
+    for u in range(1, 640, 1):
+        for v in range(1, 480, 1):
+            number = v * 640 + u
+            points[number, 2] = img[v, u] / factor
+            points[number, 0] = (u - cx) * points[number, 2] / fx
+            points[number, 1] = (v - cy) * points[number, 2] / fy
     pcd_new = o3d.geometry.PointCloud()
     pcd_new.points = o3d.utility.Vector3dVector(points)
     return pcd_new
 
 
-"""Set constants"""
-dataset_path = 'C:/Users/maksn/Desktop/rgbd_dataset_freiburg1_xyz/'
-image_num = 84
-w = 480
-h = 640
-"""Set the camera"""
-camera_parameters_Freiburg1 = [591.1, 590.1, 	331.0,	234.0, -0.0410, 0.3286,	0.0087,	0.0051, -0.5643]
-camera_data_file = 'groundtruth.txt'
-camera = camera_from_tum(dataset_path, camera_parameters_Freiburg1, camera_data_file, image_num)
-"""Get the image, create clouds both ways"""
-image_depth_png = png_image_depth_from_tum(dataset_path, image_num)
-im_d = np.asarray(image_depth_png)
-cloud = pcd_from_o3d_depth_image(o3d.geometry.Image(im_d), camera)
-cloud_new = pcd_from_depth_image(im_d, camera_parameters_Freiburg1)
-o3d.visualization.draw_geometries(cloud_new)
-"""Visualizer"""
-vis = o3d.visualization.Visualizer()
-vis.create_window(height=h, width=w)
-vis.add_geometry(cloud)
-ctr = vis.get_view_control()
-init_param = ctr.convert_to_pinhole_camera_parameters()
-init_param.intrinsic = camera.intrinsic
-init_param.extrinsic = camera.extrinsic
-ctr.convert_from_pinhole_camera_parameters(init_param)
-vis.run()
+def visualizer_set_camera(vis_object, camera_object):
+    ctr = vis_object.get_view_control()
+    init_param = ctr.convert_to_pinhole_camera_parameters()
+    init_param.intrinsic = camera_object.intrinsic
+    init_param.extrinsic = camera_object.extrinsic
+    ctr.convert_from_pinhole_camera_parameters(init_param)
+
+
+if __name__ == "__main__":
+    """Set constants"""
+    dataset_path = 'C:/Users/maksn/Desktop/rgbd_dataset_freiburg1_xyz/'
+    image_num = 84
+    h = 480
+    w = 640
+    """Set the camera"""
+    camera_parameters_Freiburg1 = [591.1, 590.1, 	331.0,	234.0, -0.0410, 0.3286,	0.0087,	0.0051, -0.5643]
+    camera_data_file = 'groundtruth.txt'
+    camera = camera_object_from_tum_data(dataset_path, camera_parameters_Freiburg1, camera_data_file, image_num)
+    """Get the image, create clouds both ways"""
+    image_depth_png = png_image_from_tum_data(dataset_path, 'depth', image_num)
+    im_d = np.asarray(image_depth_png)
+    cloud = point_cloud_from_o3d_depth_image(o3d.geometry.Image(im_d), camera)
+    cloud_new = point_cloud_from_depth_image(im_d, camera_parameters_Freiburg1)
+    """Visualizer"""
+    vis = o3d.visualization.Visualizer()
+    vis.create_window(window_name='Point cloud from depth image {}, created via open3d'.format(image_num),
+                      width=camera.intrinsic.width,
+                      height=camera.intrinsic.height)
+    vis.add_geometry(cloud)
+    # visualizer_set_camera(vis, camera)
+    vis.run()

```

Comment:
`o3d.visualization.draw_geometries([list of pcds])` will do everything for you, no Visualizer needed